File: C++/to_base_n.cpp
#include <iostream>
#include <math.h>
using namespace std;

void base_to_n(int base, int n);

int main(void)
{
    int f=0;
    double x,y;
    cout << "Please enter base number and system number below (Use space to separate them):\n";
    do
    {
        f=1;
        cin >> x >> y;
        if(x<0||y<=0||floor(x)!=x||floor(y)!=y||y>=36)
            f=0;
        if(f==0)
            cout << "You have input the wrong number, please try again:\n";
    }
    while(f==0);
    base_to_n((int)x, (int)y);
}

void base_to_n(int base, int n)
{
    int temp;
    temp = base % n;
    if(base >= n)
    {
        base = base / n;
        base_to_n(base, n);
    }
    if(temp < 10)
        cout << temp;
    else
        cout << (char)(55 + temp);
}

File: Ruby/to_base_n.rb
def main()
	begin
		f = 1
		puts "Please enter base number:"
		x = gets
		puts "Please enter system number:"
		y = gets
		x = x.to_i
		y = y.to_i
		if (x<0||y<=0||y>=36) then
			f = 0
		end
		if f==0 then
			puts "You have input the wrong number, please try again:"
		end
	end while f==0
	print "The result is:"
	to_base_n(x,y)
	puts ""
	puts "Press any key to exit..."
	x = gets
end

def to_base_n(base,n)
	temp = 0
	temp = base % n
	if base>=n then
		base = (base/n).floor.to_i
		to_base_n(base,n)
	end
	if temp<10 then
		putc (48+temp.to_i).chr
	else
		putc (55+temp.to_i).chr
	end
	return;
end

main()